virttest/virsh.py

```diff
@@ -1577,8 +1577,16 @@ def undefine(name, options=None, **dargs):
     :return: CmdResult object
     """
     cmd = "undefine %s" % name
-    if options is not None:
-        cmd += " %s" % options
+
+    vmxml = dumpxml(name).stdout_text
+    if "nvram" in vmxml:
+        if options is None or "--nvram" not in options:
+            cmd += " --nvram"
+        else:
+            cmd += " %s" % options
+    else:
+        if options is not None:
+            cmd += " %s" % options
```

Comment:
As --nvram does not have side effect even on seabios guest, we can safely use it on both seabios and uefi guests. So let's remove above special checking. Thanks.

Reply:
Updated